src/commands/distribute/releases/edit.ts

```diff
@@ -70,23 +78,55 @@ export default class EditReleaseCommand extends AppCommand {
       handleHttpError(error, false, "failed to load release details");
     }
 
+    let enabled: boolean = null
+    if(this.state === "enabled" || this.state === "disabled"){
+      enabled = this.state === "enabled"
+    } 
+    const releaseNotes = await this.getReleaseNotesString()
+
     try {
-      debug(`Updating release state to "${state}"`);
+      debug(`Updating release`);
       const httpResponse = await out.progress(
-        `${state === "enabled" ? "Enabling" : "Disabling"} the release...`,
+        `Updating release`,
         clientRequest((cb) =>
-          client.releasesOperations.updateDetails(releaseId, app.ownerName, app.appName, { enabled: state === "enabled" }, cb)
+          client.releasesOperations.updateDetails(releaseId, app.ownerName, app.appName, { enabled: enabled , releaseNotes: releaseNotes }, cb)
         )
       );
       if (httpResponse.response.statusCode >= 400) {
-        return failure(ErrorCodes.Exception, `failed to ${state === "enabled" ? "enable" : "disable"} the release`);
+        return failure(ErrorCodes.Exception, `failed to update the relese`);
       }
     } catch (error) {
       debug(`Failed to update the release - ${inspect(error)}`);
-      return failure(ErrorCodes.Exception, `failed to ${state === "enabled" ? "enable" : "disable"} the release`);
+      return failure(ErrorCodes.Exception, `failed to update the relese`);
     }
 
-    out.text(`Release ${releaseDetails.shortVersion} (${releaseDetails.version}) with id: ${this.releaseId} has been ${state}`);
+    out.text(`Release ${releaseDetails.shortVersion} (${releaseDetails.version}) with id: ${this.releaseId} has been updated`);
     return success();
   }
+
+  private async getReleaseNotesString(): Promise<string> {
+    if (!_.isNil(this.releaseNotesFile)) {
+      try {
+        return await Pfs.readFile(this.releaseNotesFile, "utf8");
+      } catch (error) {
+        if (error.code === "ENOENT") {
+          throw failure(ErrorCodes.InvalidParameter, `release notes file '${this.releaseNotesFile}' doesn't exist`);
+        } else {
+          throw error;
+        }
+      }
+    } else {
+      return this.releaseNotes;
+    }
+  }
+
+  private validateParameters(): void {
+    debug("Checking for invalid parameter combinations");
+    if (!_.isNil(this.releaseNotes) && !_.isNil(this.releaseNotesFile)) {
+      throw failure(ErrorCodes.InvalidParameter, "'--release-notes' and '--release-notes-file' switches are mutually exclusive");
+    }
+    if (!_.isNil(this.state) && this.state === "enabled" || this.state === "disable" ) {
```

Comment:
There's some issues here. `&&` takes precedence over `||`. I don't think this does what you want it to do. The check also needs to be inverted to make sense. And it's checking for `disable` instead of `disabled`. Let's go with this:

`if (!_.isNil(this.state) && this.state !== "enabled" && this.state !== "disabled" )`